src/activation.jl

```diff
@@ -35,6 +44,14 @@ logσ(x::Real) = -softplus(-x)
 const logsigmoid = logσ
 
 
+"""
+    hardtanh(x) = max(-1, min(1, x))
+
+Segment-wise linear approximation of tanh. Cheaper  and  more  computational  efficient version of tanh.
+"""
+hardtanh(x::Real) = oftype(x / 1,max(-1, min( 1, x)))
```

Comment:
Is there a paper that introduced `hardtanh()`?

Reply:
I am not completely sure that if the following paper introduced this, but they have used this in NLP
http://www.jmlr.org/papers/volume12/collobert11a/collobert11a.pdf

Reply:
And they have claimed that this function is computationally cheaper than tanh itself.

Reply:
Oh, it's definitely cheaper, I can tell you that.  ;)

Reading that paper, they state that the `HardTanh` was first introduced in `(Collobert, 2004)` which turns out to be the same author's thesis: http://ronan.collobert.org/pub/matos/2004_phdthesis_lip6.pdf

So we'll just cite that as the originating paper.  :)

Reply:
Ok, I will add that with the function